src/encoded/static/components/cart/cart.js

```diff
@@ -943,9 +941,16 @@ const CartTools = ({ elements, selectedTerms, selectedDatasetType, typeChangeHan
     // datasets."
     const disabledMessage = selectedDatasetType === DEFAULT_DATASET_TYPE ? 'Select dataset type to download' : '';
     const selectedType = datasetTypes[selectedDatasetType].type;
+
+    // Make a list of all the dataset types currently in the cart.
+    const usedDatasetTypes = [Object.keys(defaultDatasetType)[0]].concat(elements.reduce((types, elementAtId) => {
+        const type = atIdToType(elementAtId);
+        return types.includes(type) ? types : types.concat(type);
+    }, []));
```

Comment:
I don't quite follow the distinction between "usedDatasetTypes" and "actualDatasetTypes" - why can't we compute "actualDatasetTypes" here and use it for both Cart components? CartTypeSelector needs the "all" type?

Reply:
Yeah, the only difference is that `usedDatasetTypes` includes "all" while `actualDatasetTypes` does not.